### PR TITLE
DCOS-2274: Show error message to make it clear that command is not valid

### PR DESCRIPTION
**Pull request by truskovskiyk**

### CODE REVIEW

**Comment by tamarrow** [comment]
Rebase to fix the build problem in the tests

**Comment by ssk2** [comment]
Please rebase on top of master and squash your commits.

File: cli/dcoscli/marathon/main.py [hunk 1]
```diff
@@ -105,17 +105,23 @@
...
 
+@handle_docopt_error
 def main():
     try:
         return _main()
     except DCOSException as e:
         emitter.publish(e)
         return 1
+    except docopt.DocoptExit as e:
+        emitter.publish("Command Not Recognised. Please see Usage")
```
Comment (ssk2) [inline cli/dcoscli/marathon/main.py]: You can remove "Please see Usage" from the error message since this is explicit (it prints Usage below).

Comment (ssk2) [inline cli/dcoscli/marathon/main.py]: Can we add a new line before the Usage information is printed too please?

File: cli/dcoscli/main.py [hunk 2]
```diff
@@ -137,3 +138,24 @@ def signal_handler(signal, frame):
     emitter.publish(
         errors.DefaultError("User interrupted command with Ctrl-C"))
     sys.exit(0)
+
+
+def handle_docopt_error(func):
```
Comment (ssk2) [inline cli/dcoscli/main.py]: Rename this function - it's not handling the error, more like decorating it. Perhaps `def decorate_docopt_usage(func)`

File: cli/dcoscli/marathon/main.py (hunk 1 above)
Comment (ssk2) [inline cli/dcoscli/marathon/main.py]: Actually, if we're wrapping this function, why is this exception block here?

File: cli/dcoscli/main.py [hunk 3]
```diff
@@ -137,3 +138,24 @@ def signal_handler(signal, frame):
...
+    :return: wrapped function
+    :rtype: function
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            result = func(*args, **kwargs)
+        except docopt.DocoptExit as e:
+            emitter.publish("Command Not Recognised. Please see Usage")
```
Comment (ssk2) [inline cli/dcoscli/main.py]: You can remove "Please see Usage" from the error message since this is implicit (it prints Usage below).

Comment (ssk2) [inline cli/dcoscli/main.py]: Can we add a new line before the Usage information is printed too please?

File: cli/dcoscli/config/main.py [hunk 4]
```diff
@@ -35,11 +35,13 @@
                   subcommand, util)
 from dcos.errors import DCOSException
 from dcoscli import analytics
+from dcoscli.main import handle_docopt_error
 
 emitter = emitting.FlatEmitter()
 logger = util.get_logger(__name__)
 
 
+@handle_docopt_error
```
Comment (ssk2) [inline cli/dcoscli/config/main.py]: Why not wrap the inner function `_main()`?

File: cli/dcoscli/main.py [hunk 5]
```diff
@@ -137,3 +138,24 @@ def signal_handler(signal, frame):
     emitter.publish(
         errors.DefaultError("User interrupted command with Ctrl-C"))
     sys.exit(0)
+
+
+def handle_docopt_error(func):
+    """Handle DocoptExit exception
```
Comment (ssk2) [inline cli/dcoscli/main.py]: And update this text too.

**Comment by tamarrow** [comment]
what's the status on this? @ssk2 @truskovskiyk

**Comment by truskovskiyk** [comment]
@ssk2 @tamarrow PR updated please review, all tests passed

**Comment by tamarrow** [comment]
one small comment, rebase your commits and then lgtm.

**Comment by truskovskiyk** [comment]
@tamarrow done

**Pull request merged**